components/core/src/clp_s/search/ColumnDescriptor.hpp

```diff
@@ -62,14 +58,47 @@ class DescriptorToken {
      * @return Whether this token is equal to the given token
      */
     bool operator==(DescriptorToken const& rhs) const {
-        // Note: we only need to compare the m_token field because m_regex and m_wildcard are
-        // derived from m_token.
-        return m_token == rhs.m_token;
+        return m_token == rhs.m_token && m_wildcard == rhs.m_wildcard;
```

Comment:
Yes -- the current implementation now receives an escaped token, unescapes it, and sets relevant flags. Since the implementation unescapes the string it stores internally the flags need to be used to differentiate some cases.

In particular if you construct a DescriptorToken with `\*` it will internally store `*` and the m_wildcard flag will be false, whereas if you construct it with `*` it will again store `*` but the m_wildcard flag will be true.